server/src/food/routes/FoodItemCategory.route.ts

```diff
@@ -0,0 +1,44 @@
+import { Router } from "express";
+import { ModelValidationHandler, ValidationGroup } from "../../app/index";
+import { AuthValidationHandler } from "../../auth/index";
+import { FoodItemCategory, FoodItemCategoryHandler } from "../index";
+
+export class FoodItemCategoryRoute {
+
+  private _objToValidate: FoodItemCategory;
+
+  constructor (
+    private _router: Router,
+    private _fooditemCategoryHandler: FoodItemCategoryHandler,
+    private _authValidationHandler: AuthValidationHandler,
+    private _modelValidationHandler: ModelValidationHandler
+  ) {
+    this._objToValidate = new FoodItemCategory();
+  }
+
+  public createRoutes (): Router {
+    this._router.get("/",
+      this._authValidationHandler.checkToken,
+      this._fooditemCategoryHandler.getAll);
+
+    this._router.get("/:id",
+      this._authValidationHandler.checkToken,
+      this._fooditemCategoryHandler.getById);
+
+    this._router.put("/:id",
+      this._authValidationHandler.checkToken,
```

Comment:
Is everybody allowed to create, update and delete categories? Maybe it's better to provide the users with a consistent predefined set of categories and only allow admins to update this list. What do you think?

Reply:
that's a good idea. i will add this check.

Reply:
what is with fooditem unit and fooditem template? should we restrict this too?

Reply:
I'm not sure but at the moment I would say no. Let's keep this in mind